### ISSUE TITLE
Normalize marcgac

### ISSUE THREAD

**Issue by justinlittman** [sul-dlss/dor-services-app#2805]
We currently map marcgac as is, which may include trailing dashes so that the code is padded to 7 characters. For example, `n-us---`.

Arguably those trailing dashes are not part of the code and should be removed as part of mapping so that only valid marcgac codes are in the cocina model.

Note that this is likely to require mods normalization for roundtrip testing.

**Comment by justinlittman** [comment]
Also, once this is done handling for trailing dashes can be removed from GeographicBuilder in dor_indexing_app.
